gr-digital/python/digital/qa_ofdm_chanest_vcvc.py

```diff
@@ -301,13 +301,14 @@ def run_flow_graph(sync_sym1, sync_sym2, data_sym):
                 rand_range(min_chan_ampl, max_chan_ampl) *
                 numpy.exp(1j * rand_range(0, 2 * numpy.pi))
                 for x in range(fft_len)]
-            src = blocks.vector_source_c(tx_data, False, fft_len)
+            src = blocks.vector_source_c(tx_data, True, fft_len)
             chan = blocks.multiply_const_vcc(channel)
             noise = analog.noise_source_c(analog.GR_GAUSSIAN, wgn_amplitude)
             add = blocks.add_cc(fft_len)
             chanest = digital.ofdm_chanest_vcvc(sync_sym1, sync_sym2, 1)
             sink = blocks.vector_sink_c(fft_len)
-            top_block.connect(src, chan, (add, 0), chanest, sink)
+            head = blocks.head(gr.sizeof_gr_complex * fft_len, 1)
+            top_block.connect(src, chan, (add, 0), chanest, head, sink)
```

Comment:
So you want to add a `head` block to make sure the flowgraph finishes. The issue would be that the `noise` block, in parallel to `src` does not have a "generate a finite amount of samples" option. The scheduler might hang on that. Thus, the qa test doesn't pass.
tbh, I'd prefer a solution that doesn't require another block that just forces the flowgraph to stop. Would it be possible to use a `fast_noise_source` instead of the current noise source? The `fast_noise_source` block should be a GRC only block that basically wraps `blocks.vector_source_c(np.random.normal(...), [True/False], fft_len)`.

Reply:
It also makes the vector source repeat to keep pushing samples, and using the head instead of the vector source to signal done acts to make sure there are enough samples at the end of flowgraph before it is killed.

Reply:
Yeah. And I suggest to take a different approach here because it would be good to have a test with exact input `x` to produce exact output `y`. If that's not the case, this might be a good indicator that something is broken. I know that `np.random....` or some other random function interferes with that.

Reply:
Gotcha - the general flow works if I change the noise_source to a vector source with np.random.normal vector as its values.
I think the underlying issue is the single value items for the noise source upstream from the add block requires more samples to be pushed through the pipeline before the vector source with 1 vector says it is done.